src/redux/books/bookSlice.js

```diff
@@ -1,42 +1,109 @@
-import { createSlice } from '@reduxjs/toolkit';
+/* eslint-disable import/no-extraneous-dependencies */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/PWiB57pJPPFLc1okU4BW/books';
 
 const initialState = {
-  books: [
-    {
-      itemId: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      itemId: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      itemId: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
+  isLoading: false,
+  isError: false,
+  isSuccessful: false,
+};
+
+const changeToObjectData = (data) => {
+  const newDataArray = [];
+  data.forEach((element) => {
+    const newObject = {
+      id: element[0],
+      title: element[1][0].title,
+      author: element[1][0].author,
+      category: element[1][0].category,
+    };
+    newDataArray.push(newObject);
+  });
+  return newDataArray;
 };
 
-const booksSlice = createSlice({
-  name: 'books',
+export const getBooks = createAsyncThunk('book/getBooksData', async () => {
+  try {
+    const dataStream = await axios(url);
+    let data = Object.entries(dataStream.data);
+    data = changeToObjectData(data);
+    return data;
+  } catch (error) {
+    return error;
+  }
+});
+
+export const addBooks = createAsyncThunk(
+  'book/addBooks',
+  async ({ id, title, author }) => {
+    try {
+      const dataStream = await axios.post(url, {
+        itemId: id,
+        title,
+        author,
+        category: 'Non fiction',
+      });
+      return dataStream;
+    } catch (error) {
+      return error;
+    }
+  },
+);
```

Comment:
- [ ] I'm getting the following error trying to create a new book:

```json
{"error":{"status":400,"message":"'item_id' is required."}}
```

it seems to be a typo in `itemId` it should be `item_id`. Please consider applying this change! 🤓 🔧